### PR TITLE
[metricbeat] gcp: add distribution type metrics

### LINKED ISSUE #17975: [Metricbeat] Add distribution type metrics into googlecloud module
Googlecloud has a time series type `[distribution](https://cloud.google.com/monitoring/api/ref_v3/rest/v3/TimeSeries#Distribution)` which contains summary statistics for a population of values. It optionally contains a histogram representing the distribution of those values across a set of buckets. The summary statistics are the count, mean, sum of the squared deviation from the mean, the minimum, and the maximum of the set of population of values. The histogram is based on a sequence of buckets and gives a count of values that fall into each bucket. The boundaries of the buckets are given either explicitly or by formulas for buckets of fixed or exponentially increasing widths.

https://github.com/elastic/beats/pull/17946 removed collecting distribution metrics from `googlecloud` module. This issue is to track the work for adding distribution metrics back with ES histogram support.

### PR DESCRIPTION
**Pull request by ghost**
## What does this PR do?

Adds GCP distribution type metrics.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

- Closes #17975

### CODE REVIEW

File: x-pack/metricbeat/module/gcp/distribution.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+	// Count must be equal to the sum of values in BucketCounts.
+	return d.Count == sum
+}
+
+func calcExplicitUpperBound(bucket *distribution.Distribution_BucketOptions_Explicit, i int) float64 {
+	return bucket.Bounds[i]
+}
+
+func calcExponentialUpperBound(bucket *distribution.Distribution_BucketOptions_Exponential, i int) float64 {
```
Comment (kaiyan-sheng) [inline x-pack/metricbeat/module/gcp/distribution.go:34]: Is there any documentation we can add or comments in the code to explain the different histograms (explicit, exponential, linear...) and what are the bucket options available from GCP?

**Review by kaiyan-sheng — COMMENTED** [review]
Great work adding the distribution type!! Are you planning to enable some distribution type metric collection here in this PR for some of the metricsets?

**Comment by gpop63** [comment]
I added comments regarding distribution histograms, and distribution metrics for `dataproc`, `loadbalancing` and `pubsub`.

**Review by kaiyan-sheng — COMMENTED** [review]
Hey @gpop63 sorry I forgot to review this again. Could you add a changelog entry for this feature please? Thank you!!!

**Review by kaiyan-sheng — APPROVED** [review]
Lets add the changelog in a separate PR.

**Pull request merged**